apps/designer/app/designer/features/sidebar-left/panels/pages/pages.tsx

```diff
@@ -86,7 +96,19 @@ const PagesPanel = ({
 
   return (
     <>
-      <Header title="Pages" onClose={onClose} />
+      <Header
+        title="Pages"
+        suffix={
+          <>
+            {onNewPage && (
+              <IconButton size="2" onClick={onNewPage} aria-label="New Page">
```

Comment:
I would inline a callback to avoid leaking the click event to a handler that shouldn't know about implementation detail and to make semantics of onNewPage make more sense because that callback could be called without a click event from some other logic, just theoretically.